### PR TITLE
disallow insertion of point with tag 0;

### PR DESCRIPTION
**Pull request by harsha-simhadri**
update tools to insert non-zero tag;
GT ignores location with 0 tags

### CODE REVIEW

**Review by rakri — APPROVED** [review]
Good to go.

File: include/index.h [hunk 1]
```diff
@@ -192,11 +192,11 @@ namespace diskann {
...
     DISKANN_DLLEXPORT size_t search_with_tags(const T *query, const uint64_t K,
                                               const unsigned L, TagT *tags,
-                                              float *           distances,
+                                              float            *distances,
                                               std::vector<T *> &res_vectors);
 
-    // Will fail if tag already in the index
-    DISKANN_DLLEXPORT int insert_point(const T *point, const TagT tag);
+    // Will fail if tag already in the index or if tag=0.
+    DISKANN_DLLEXPORT int insert_point(const T *point, const TagT &tag);
```
Comment (theantony) [inline include/index.h:199]: I think we're limiting `TagT` to primitive types (there are asserts to that effect in e.g. `natural_number_map.h` and `index` template is instantiated only for a few primitive types). Given that, reference for tag just adds unnecessary level of indirection.

File: tests/utils/compute_groundtruth.cpp [hunk 2]
```diff
@@ -278,8 +279,7 @@ inline void load_bin_as_float(const char *filename, float *&data, size_t &npts_u
   uint64_t end_id = (std::min)(start_id + PARTSIZE, (uint64_t) npts_i32);
   npts_u64 = (uint64_t) end_id - (uint64_t) start_id;
```
Comment (theantony) [inline tests/utils/compute_groundtruth.cpp:280]: (From the previous PR) Are the two casts here necessary, the two variables already have `uint64_t` type?

**Review by theantony — APPROVED** [review]
My suggestions are minor, feel free to merge as-is.

File: include/index.h (hunk 1 above)
Comment (harsha-simhadri) [reply include/index.h:199]: I can remove the reference for now. Is it easy to generalize the nnmap.h to string tag types?

Comment (theantony) [reply include/index.h]: AFAIR the main thing would be to make sure that primitive types are passed by value and `std::string` etc. by reference. Should be relatively straightforward, but I didn't want to spent time on it when creating the class as it wasn't necessary.

**Pull request merged**